### DISCUSSION TITLE
Enable running local python code/models

### DISCUSSION THREAD

**Discussion by ismailsimsek** [dbt-labs/dbt-core#9741] [Ideas]
Currently dbt allows running python inside destination system/database(snowpar, pyspark). But for some cases it is useful to run local python scripts 

for example `web-api->local_dbt_python_model(reads data from api)->converts to pandas.dataframe->save_to_destination_table`

for this probably `submit_local_python_job` method could be created. similar to `submit_python_job`

https://github.com/dbt-labs/dbt-adapters/blob/2a99e36a7479ec23f40456e89377419aae7388e7/dbt/adapters/base/impl.py#L1423

**Comment by ismailsimsek** [comment]
will be great to add this to core

**Comment by ismailsimsek** [comment]
this is now possible with [opendbt](https://github.com/memiiso/opendbt )  project. 

[explained here](https://github.com/memiiso/opendbt/blob/main/docs/EXAMPLES.md#execute-python-model-locally)

**Comment by candalfigomoro** [comment]
The ability to run local Python models (without Spark) would be especially important since dbt-fal (https://github.com/fal-ai/dbt-fal) has been discontinued.

**Reply by ismailsimsek** [reply]
@candalfigomoro this is implemented now with [opendbt](https://github.com/memiiso/opendbt) project.

Documented here: https://github.com/memiiso/opendbt/blob/main/docs/EXAMPLES.md#execute-python-model-locally
Unit test (Example):  https://github.com/memiiso/opendbt/blob/0da7344b1eecb4d91582d33c5aab78831cb59959/tests/test_executepython_materialization.py#L11

**Reply by candalfigomoro** [reply]
@ismailsimsek Thanks, I looked at the examples but it seems cumbersome compared to having a simple native implementation in dbt.

**Comment by ismailsimsek** [comment]
Related https://github.com/dbt-labs/dbt-adapters/pull/261

**Comment by mrcool4** [comment]
We are also interested in this feature
